### ISSUE TITLE
Support for Cry Sensor with Arlo Baby

### ISSUE THREAD

**Issue by chansearrington** [twrecked/hass-aarlo#149]
Thanks a ton for this integration and specifically for supporting Arlo Baby!

I'm updating my config right now to add temperature/humidity/air_quality as we speak. (just a heads up I don't see that in the documentation)

Is there any chance that you can add the crying sensor in addition to the sound sensor?

we've been looking for a way to get a better notification when our kids crying and this would enable us to plumb it into alexa notifications and/or chromecast

**Comment by sebr** [comment]
I did a bit of debugging on this last night. Unfortunately I couldn't find any messages on the Arlo event stream which allows the consumer to determine that the audio detected event is a "cry" event.

I think the same problem exists for Arlo Smart notifications - see #110

**Comment by chansearrington** [comment]
Ok, well that stinks :-)

Thanks for looking into it.

Do you have an Arlo Baby? would it help if I send you my event stream?

**Comment by sebr** [comment]
Indeed, it does stink!

I do have an Arlo Baby device. Feel free to dig around your event stream and if you see anything of interest pop it in here!

**Comment by twrecked** [comment]
If you spot something to indicate crying in the log we can maybe use that. I just added some code to add the object type - vehicle, person, animal or other - to the last capture sensor based on the last video recording made.

**Comment by chansearrington** [comment]
@twrecked how/where do I access my "log"?

**Comment by twrecked** [comment]
Ah! I've been trying to improve the documentation so I hope this link works: https://github.com/twrecked/hass-aarlo#other-debugging

But basically, you need to add some configuration and restart your system.

```yaml
logger:
  default: info
  logs:
    pyaarlo: debug

aarlo:
    # current config here
    packet_dump: True
```

Then look in `/config/home-assistant.log` and `/config/.aarlo/packets.dump` for something to do with crying or audioDetected or audioNotification or audioSomething. 

Do you record a video when the crying starts? The video library includes the reason the video was started, we may be able to hook into that.

**Comment by chansearrington** [comment]
Thanks, @twrecked! I just looked through the dump and...

Is this the sensor and state? (from the dump file)

enabled = true
triggered = false

`  'properties': [ { 'activityState': 'idle',
                    'audioAnalytics': { 'babyCryDetection': { 'enabled': True,
                                                                                    'triggered': False}},`

**Comment by sebr** [comment]
@chansearrington awesome stuff! I just found this in my packet dump:

```
{ 'action': 'is',
  'from': 'CAMERA_ID',
  'properties': {'audioAnalytics': {'babyCryDetection': {'triggered': True}}},
  'resource': 'cameras/CAMERA_ID',
  'transId': 'CAMERA_ID!transid'}
```

**Comment by Siege36** [comment]
Interesting. I have had an Arlo baby for 2 years and didn’t know this. Recently had twins if you need some test cry detection!

**Comment by chansearrington** [comment]
I have a 4-year-old, 2-year-old and 2 month old. We use this like crazy :-)

**Comment by twrecked** [comment]
That we can use!

Is there a corresponding `{'triggered': False}`? If not I'll just create a timer like I did for the doorbell.

**Comment by chansearrington** [comment]
@twrecked yes, I had  'triggered': False as the value in my original post above.

**Comment by twrecked** [comment]
Excellent. I can definitely hook into those. I'll try and get a test build out last this week.

**Comment by twrecked** [comment]
I just checked the initial work into this branch - https://github.com/twrecked/hass-aarlo/tree/arlo-baby-light

The detection is working at the camera level I just need to create the sensor type and then I'll create an official release.

If you are comfortable using a development branch then give it a go.

**Comment by chansearrington** [comment]
Whoohoo!!! Can’t wait to try this. I’ll give it a go this weekend when I get back in town. Thanks a ton!

**Comment by twrecked** [comment]
It's now in the latest release. Just make sure to upgrade HACS to the latest release before upgrading Aarlo. There was a bug in HACS that flattened the directory structure out.

**Comment by mikesalz** [comment]
@twrecked Is the cry sensor type now in the prod release or still in a dev branch?  I have v0.6.14 and received the following error when I added - cry to the monitored conditions list:

Invalid config for [sensor.aarlo]: value is not allowed @ data['monitored_conditions'][2]. Got 'cry'. (See ?, line ?).

**Comment by mikesalz** [comment]
@twrecked Do we know if anyone has successfully tested the cry detection?  I enabled the cry detection and then played an obnoxious youtube video of a baby crying.  My blood pressure went up, sensor never triggered.  Thoughts?  Thanks!

**Comment by sebr** [comment]
It's not worked for me (and there is _a lot_ of crying that goes on in my house 😂). I've not had the opportunity to debug it though.

**Comment by twrecked** [comment]
Does it work on the web interface? Can you check for `babyCryDetection` in packet dumps, I could be reading that out wrong.

**Comment by mikesalz** [comment]
@twrecked Thanks for working on this!  I installed v0.6.14.4 tonight and made my kids pretend to cry in their rooms.  The sound sensor picks it up with no problem, but the cry sensor still does not work.

**Comment by twrecked** [comment]
Damn. Any chance you can get some packet traces around the time it happens? I'll double check my latest code and make sure it still picking them up.

**Comment by twrecked** [comment]
I found the issue. I'll check in a fix.

**Comment by mikesalz** [comment]
I installed v0.6.14.5 this morning.  Just checked out the sensors.  I have two Arlo baby cameras.  One camera seems to be permanently detecting crying while the room is silent.  The other is not detecting crying at all, even with that obnoxious youtube crying baby video playing at full volume.

**Comment by twrecked** [comment]
Can you look through your debug and see if you have any lines that look like this - the `babyCryDetection` is the bit you're looking for:

```yaml
2020-03-13 10:39:50 DEBUG (MainThread) [pyaarlo] sending cameras/XXXXXXXXXXXX to XXXXXXXXXXXX
2020-03-13 10:39:50 DEBUG (MainThread) [pyaarlo] Front Door Camera CAMERA got one cameras/XXXXXXXXXXXX
2020-03-13 10:39:50 DEBUG (MainThread) [pyaarlo] set:ArloCamera/XXXXXXXXXXXX/babyCryDetection=True
```
